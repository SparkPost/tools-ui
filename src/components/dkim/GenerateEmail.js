import React from 'react';
import { ActionButton } from '../button/Button';

export default (props) => (
  <div className='panel panel--accent'>
    <div className='panel__body paddingTop--xl paddingBottom--xl text--center'>
      <h4>To get started, click on the button below to generate an email address.</h4>
      <ActionButton type='blue' action={props.generate}>Generate Email Address</ActionButton>
    </div>
  </div>
);
