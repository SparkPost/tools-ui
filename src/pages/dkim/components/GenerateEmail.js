import React from 'react';
import { Button } from 'components/button/Button';

export default (props) => (
  <span>
    {/* will use for the major version upgrade in the following days/weeks FE-1350 */}
    {/* <div style={{ background: '#fa6423', color: '#fff', padding: '15px' }}>
      Sorry, this tool is down for a scheduled maintenance right now. Please check back in a little bit.
    </div> */}
    <div className='panel panel--accent'>
      <div className='panel__body paddingTop--xl paddingBottom--xl text--center'>
        <h4>To get started, click on the button below to generate an email address.</h4>
        <Button type='blue' action={props.generate} track={true}>Generate Email Address</Button>
      </div>
    </div>
  </span>
);
