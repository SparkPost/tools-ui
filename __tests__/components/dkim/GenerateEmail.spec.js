import React from 'react';
import renderer from 'react-test-renderer';

import GenerateEmail from 'components/dkim/GenerateEmail';

describe('GenerateEmail component', () => {

  const generate = () => {};

  test('should render correctly with a generate function', () => {
    expect(renderer.create(<GenerateEmail generate={generate} />)).toMatchSnapshot();
  });

});
