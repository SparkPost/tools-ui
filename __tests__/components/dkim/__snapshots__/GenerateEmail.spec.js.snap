exports[`GenerateEmail component should render correctly with a generate function 1`] = `
<span>
  <div
    className="panel panel--accent">
    <div
      className="panel__body paddingTop--xl paddingBottom--xl text--center">
      <h4>
        To get started, click on the button below to generate an email address.
      </h4>
      <button
        className="button button--blue"
        onClick={[Function]}>
        Generate Email Address
      </button>
    </div>
  </div>
</span>
`;
